File: backend/apiserver/routes/mission.go
//
// Copyright © 2019 Sunho Kim. All rights reserved.
//

package routes

import (
	"gorani/middles"
	"gorani/models"

	"github.com/sunho/webf/servs/dbserv"

	"github.com/sunho/dim"
)

type Mission struct {
	DB *dbserv.DBServ `dim:"on"`
}

func (c *Mission) Register(d *dim.Group) {
	d.Use(&middles.AuthMiddle{})
	d.GET("", c.Get)
}

func (m *Mission) Get(c *models.Context) error {
	class, err := c.User.GetClass()
	if err != nil {
		return err
	}
	mission, err := class.GetCurrentMission()
	if err != nil {
		return err
	}
	return c.JSON(200, mission)
}

File: frontend/ios/App/Models/Mission.swift
//
// Copyright © 2019 Sunho Kim. All rights reserved.
//

import Foundation
import FirebaseFirestore

struct Class: Codable {
    let name: String
    let mission: Mission?
}

struct Mission: Codable {
    let bookId: String?
    let id: String
    let message: String
    let due: Timestamp
    let chapters: [String]?
}
